### PR TITLE
Add configuration options for Aggregation creation

### LINKED ISSUE #1444: [Metrics SDK] Add configuration options for Aggregation creation
As of now, the aggregation use default configuration - Eg Histogram aggregation uses default values of buckets and is not configurable. It should be possible for application developers to specify the configuration options while creating aggregation.

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1444 (issue)

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
Thanks for the PR. Changes are nicely done. One suggestion - Should we expose the config api in the SDK and not the API. As of now, the instrumentation library shouldn't be able to specify the metrics configuration, it should be done only by SDK. There have been some discussions earlier to expose the config through API - [Configuring histograms in the API (hints?)](https://github.com/open-telemetry/opentelemetry-specification/issues/2229), [Add the metrics API Hint](https://github.com/open-telemetry/opentelemetry-specification/pull/1753) - but was not added to specs. I think we should add the support through SDK (using View API?)

**Comment by esigo** [comment]
> Thanks for the PR. Changes are nicely done. One suggestion - Should we expose the config api in the SDK and not the API. As of now, the instrumentation library shouldn't be able to specify the metrics configuration, it should be done only by SDK (through View API?). There have been some discussions earlier to expose the config through API - [Configuring histograms in the API (hints?)](https://github.com/open-telemetry/opentelemetry-specification/issues/2229), [Add the metrics API Hint](https://github.com/open-telemetry/opentelemetry-specification/pull/1753) - but was not added to specs.

@lalitb thanks, removed from api.

File: sdk/include/opentelemetry/sdk/metrics/view/view.h [hunk 1]
```diff
@@ -24,12 +26,15 @@ class View
   View(const std::string &name,
        const std::string &description   = "",
        AggregationType aggregation_type = AggregationType::kDefault,
+       nostd::shared_ptr<AggregationConfig> aggregation_config =
```
Comment (lalitb) [inline sdk/include/opentelemetry/sdk/metrics/view/view.h:29]: nit  - we can use `std::shared_ptr` in sdk surface.

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for the PR.

File: sdk/include/opentelemetry/sdk/metrics/view/view.h (hunk 1 above)
Comment (esigo) [reply sdk/include/opentelemetry/sdk/metrics/view/view.h]: done

**Pull request merged**